src/Utilities/FlowAnalysis/Analysis/PropertySetAnalysis/PropertySetAnalysis.cs

```diff
@@ -0,0 +1,77 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using Analyzer.Utilities.Extensions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.FlowAnalysis;
+using Microsoft.CodeAnalysis.FlowAnalysis.DataFlow;
+using Microsoft.CodeAnalysis.FlowAnalysis.DataFlow.PointsToAnalysis;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Analyzer.Utilities.FlowAnalysis.Analysis.PropertySetAnalysis
+{
+    using PropertySetAnalysisData = IDictionary<AbstractLocation, PropertySetAbstractValue>;
+    using InterproceduralBinaryFormatterAnalysisData = InterproceduralAnalysisData<IDictionary<AbstractLocation, PropertySetAbstractValue>, PropertySetAnalysisContext, PropertySetAbstractValue>;
+    using PropertySetAnalysisDomain = MapAbstractDomain<AbstractLocation, PropertySetAbstractValue>;
+    using PointsToAnalysisResult = DataFlowAnalysisResult<PointsToBlockAnalysisResult, PointsToAbstractValue>;
+
+    /// <summary>
+    /// Dataflow analysis to track <see cref="PropertySetAbstractValue"/> of <see cref="AbstractLocation"/>/<see cref="IOperation"/> instances.
+    /// </summary>
+    internal partial class PropertySetAnalysis : ForwardDataFlowAnalysis<PropertySetAnalysisData, PropertySetAnalysisContext, PropertySetAnalysisResult, PropertySetBlockAnalysisResult, PropertySetAbstractValue>
+    {
+        public static readonly PropertySetAnalysisDomain BinaryFormatterAnalysisDomainInstance = new PropertySetAnalysisDomain(PropertySetAbstractValueDomain.Default);
+
+        private PropertySetAnalysis(PropertySetAnalysisDomain analysisDomain, PropertySetDataFlowOperationVisitor operationVisitor)
+            : base(analysisDomain, operationVisitor)
+        {
+        }
+
+        public static ImmutableDictionary<OperationMethodKey, PropertySetAbstractValue> GetOrComputeHazardousParameterUsages(
+            ControlFlowGraph cfg,
+            Compilation compilation,
+            ISymbol owningSymbol,
+            string typeToTrackMetadataName,
+            bool isNewInstanceFlagged,
+            string propertyToSetFlag,
+            bool isNullPropertyFlagged,
+            ImmutableHashSet<string> methodNamesToCheckForFlaggedUsage,
+            InterproceduralAnalysisKind interproceduralAnalysisKind = InterproceduralAnalysisKind.ContextSensitive,
```

Comment:
>InterproceduralAnalysisKind.ContextSensitive [](start = 70, length = 44)

Given you are using context-sensitive interprocedural analysis, you probably need to unit tests for interprocedural stuff:
1. No diagnostic: You create a formatter, pass it to another method, which sets it Binder to a non-null binder, and then the caller invokes banned method on it.
2. Diagnostic: Complementary test would create a formatter with a non-null binder, pass it as an argument to a method call that assigns null to the Binder property, and the callee invokes banned method.
3. Diagnostic: You create a formatter with Binder not set, and then pass it as an argument to a method that invokes banned method on it. I think you should get a diagnostic on argument node.

Reply:
I'll remove the interprocedural for now, and save that for another day.  Even with intraprocedural DFA the rules will be useful.

Opened https://github.com/dotnet/roslyn-analyzers/issues/1853